src/backend/replication/walsender.c

```diff
@@ -2169,36 +2191,60 @@ ProcessStandbyHSFeedbackMessage(void)
  * If wal_sender_timeout is enabled we want to wake up in time to send
  * keepalives and to abort the connection if wal_sender_timeout has been
  * reached.
+ *
+ * GPDB: If wal archiving and wal archiving status streaming is enabled, we
+ * also want to wake up in time to send the archival report.
  */
 static long
 WalSndComputeSleeptime(TimestampTz now)
 {
-	long		sleeptime = 10000;	/* 10 s */
+	long		sleeptime;
+	TimestampTz wakeup_time;
+	TimestampTz w;
 
-	if (wal_sender_timeout > 0 && last_reply_timestamp > 0)
-	{
-		TimestampTz wakeup_time;
+	wakeup_time = TimestampTzPlusMilliseconds(now, 10000);
 
+	if (wal_sender_timeout > 0 && last_reply_timestamp > 0) {
```

Comment:
The starting curly bracket needs to be on its own newline as per Postgres coding conventions.
```suggestion
	if (wal_sender_timeout > 0 && last_reply_timestamp > 0)
	{
```